StringRep.hs

```diff
@@ -70,21 +70,31 @@ postProc :: [Name] -> Term -> Term
 postProc ctx (Lam (M b) t) = Lam (M b) $ postProc (b:ctx) t
 postProc ctx (Ap t1 t2) = Ap (postProc ctx t1) (postProc ctx t2) 
 postProc ctx (Const x) | Just v <- elemIndex x ctx = LocalVar v
-                        | otherwise = Const x
+                       | otherwise = Const x
 
 type Error = MS.MisoString
 
--- TODO: need to unfold the Ap back to App Term [Term] for pretty print if_then_else_, current pretty printer only works for binary operator
+printHoley :: MS.MisoString -> [B.Builder] -> B.Builder
+printHoley origin operends = goPrintHoley origin origin operends
```

Comment:
This doesn't seem to handle the situation where a mix-fix operator is partially applied, for example `(_+_ 2)`. Like the HTML printer, it should probably check that the number of supplied arguments is the same as the number of holes before attempting to print it in mixfix form.

Reply:
Fixed. Also, should we allow the partially applied mixfix operator without underline like `+ 2`? (Currently the grammar doesn't allow any identifier using the char defined in mixfix table for preventing the ambiguity, which means the parser gives an error for it instead of recognizing it as `Ap "+" 2`)

Reply:
That's fine. It would otherwise introduce too many ambiguous parses I think.